source/common/conn_pool/conn_pool_base.h

```diff
@@ -34,6 +34,10 @@ class ActiveClient : public LinkedObject<ActiveClient>,
                uint32_t concurrent_stream_limit);
   ~ActiveClient() override;
 
+  // Called after this client is added to the list associated with its state_. At this point, the
+  // client starts to contributes to the pool capacity.
+  virtual void onEnlisted() {}
```

Comment:
The reason we need to do all this work is due to proper 0-rtt support right?  Would it be easier to just defer that to the next loop to avoid having all the special casing?

Reply:
Defer connect() to the next loop for h3.